controllers/storagecluster/provider_server.go

```diff
@@ -384,7 +394,7 @@ func GetProviderAPIServerService(instance *ocsv1.StorageCluster) *corev1.Service
 			},
 			Ports: []corev1.ServicePort{
 				{
-					NodePort:   ocsProviderServiceNodePort,
+					NodePort:   nodePort,
```

Comment:
- this is a bit unusal, either add a comment why default case (clusterip) is inverted or refactor like ...
``` go
servicePort := corev1.ServiePort{Port: ocsProviderServicePort, TargetPort: intstr.FromString("ocs-provider")}
if instance.Spec.ProviderAPIServerServiceType == corev1.ServiceTypeNodePort {
  servicePort.NodePort = ocsProviderServiceNodePort
}
// ...
Ports: []corev1.ServicePort{servicePort}
```
- I feel it's better not to set default case explicitly

Reply:
Done